src/main/java/oof/Storage.java

```diff
@@ -177,6 +192,78 @@ public void writeTaskList(TaskList taskList) {
         return arr;
     }
 
+    /**
+     * Writes Tracker objects to hard disk.
+     *
+     * @param trackerList   TrackerList of Tracker objects.
+     * @throws OofException if unable to write TrackerList.
+     */
+    public void writeTrackerList(TrackerList trackerList) throws OofException {
+        try {
+            List<Tracker> arr = trackerList.getTrackers();
+            String filename = PATH_TRACKER;
+            BufferedWriter out = new BufferedWriter(new FileWriter(filename));
+            for (int i = 0; i < trackerList.getSize(); i++) {
+                Tracker tracker = trackerList.getTracker(i);
+                out.write(tracker.toStorageString() + "\n");
+            }
+            out.close();
+        } catch (IOException e) {
+            throw new OofException("Unable to save Tracker data.");
+        }
+    }
+
+    /**
+     * Reads Tracker objects that were previously saved to hard disk.
+     *
+     * @return TrackerList that contains Tracker objects.
+     * @throws OofException if unable to read tracker.txt.
+     */
+    public TrackerList readTrackerList() throws OofException {
+        try {
+            File file = new File(PATH_TRACKER);
+            FileReader fileReader = new FileReader(file);
+            BufferedReader br = new BufferedReader(fileReader);
+            TrackerList trackerList = new TrackerList();
+            String line;
+            while ((line = br.readLine()) != null) {
+                Tracker tracker = processLine(line);
+                trackerList.addTracker(tracker);
+            }
+            return trackerList;
+        } catch (IOException | ParseException e) {
+            throw new OofException("Unable to process stored Tracker data.");
+        }
+    }
+
+    /**
+     * Processes String of line obtained from tracker.txt.
+     *
+     * @param line  String from tracker.txt.
+     * @return      Tracker object updated from data found in line.
+     */
+    private Tracker processLine(String line) throws ParseException {
+        SimpleDateFormat readFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm");
+        Date start;
+
+        String[] processed = line.split(DELIMITER_TRACKER);
+        String moduleCode = processed[ARGUMENT_FIRST];
+        String description = processed[ARGUMENT_SECOND];
+        String deadline = processed[ARGUMENT_THIRD];
+        String startDate = processed[ARGUMENT_FOURTH];
+        String lastUpdated = processed[ARGUMENT_FIFTH];
+        long timeTaken = Long.parseLong(processed[ARGUMENT_SIXTH]);
+
+        if (startDate.equals("null")) {
+            start = null;
+        } else {
+            start = readFormat.parse(startDate);
+        }
+        Date updated = readFormat.parse(lastUpdated);
+
+        return new Tracker(moduleCode, description, deadline, start, updated, timeTaken);
+    }
```

Comment:
Status of the tracker is not being processed so it is always set to not completed even if it is saved as completed

Reply:
The code has been rectified as suggested.
The tracker now updates the TaskList's Task variable (Status property) correctly.